lib/eex/test/eex_test.exs

```diff
@@ -452,6 +452,23 @@ defmodule EExTest do
       assert EExTest.Compiled.unknown() ==
                {26, {EExTest.Compiled, :unknown, 0, [file: 'unknown', line: 26]}}
     end
+
+    test "does not output compiler warning for unused variables" do
+      filename = Path.join(__DIR__, "fixtures/eex_empty_file.eex")
+
+      new_module =
+        quote do
+          defmodule EExTest.CompiledEmptyFile do
+            require EEx
+            EEx.function_from_file(:def, :empty_file_a, unquote(filename), [:bar])
+            EEx.function_from_file(:defp, :empty_file_b, unquote(filename), [:bar])
+          end
+        end
+
+      refute ExUnit.CaptureIO.capture_io(:stderr, fn ->
```

Comment:
Good catch @fertapric! I think we don't need to use `capture_io` here. Just make sure the code compiles and that's it. If there was a warning, it would be printed to the terminal, and we would have to act on it anyway.

Reply:
There are other tests using `capture_io(:stderr, ...`:

https://github.com/elixir-lang/elixir/blob/f40f21195a62f5c597c37e43e1a5dfff9b204d29/lib/eex/test/eex_test.exs#L189-L199

Reply:
Alright, so we can keep capture_io here. Indeed we have to remove `async: true`.